File: CS/Grid_BottomNewItemRow/GridDescendant/GridViewInfoDescendant.cs
using DevExpress.Data;
using DevExpress.XtraGrid.Drawing;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;

namespace Grid_BottomNewItemRow
{
    public class GridViewInfoDescendant : GridViewInfo
    {
        public GridViewInfoDescendant(GridView gridView) : base(gridView) { }

        public override int CalcRowHeight(Graphics graphics, int rowHandle, int rowVisibleIndex, int min, int level, bool useCache, GridColumnsInfo columns)
        {
            if (View.OptionsView.NewItemRowPosition == NewItemRowPosition.None && rowHandle == CurrencyDataController.NewItemRow)
                return 0;
            else
                return base.CalcRowHeight(graphics, rowHandle, rowVisibleIndex, min, level, useCache, columns);
        }


        public ScrollBarPresence secondViewVScroll = ScrollBarPresence.Hidden;

        public override ScrollBarPresence VScrollBarPresence
        {
            get
            {
                if (View.OptionsView.NewItemRowPosition == NewItemRowPosition.Bottom)
                   return secondViewVScroll;
                else
                    return base.VScrollBarPresence;
            }
        }
           
    }
}

File: VB/Grid_BottomNewItemRow/GridDescendant/GridViewInfoDescendant.vb
Imports DevExpress.Data
Imports DevExpress.XtraGrid.Drawing
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Text

Namespace Grid_BottomNewItemRow
	Public Class GridViewInfoDescendant
		Inherits GridViewInfo

		Public Sub New(ByVal gridView As GridView)
			MyBase.New(gridView)
		End Sub

		Public Overrides Function CalcRowHeight(ByVal graphics As Graphics, ByVal rowHandle As Integer, ByVal rowVisibleIndex As Integer, ByVal min As Integer, ByVal level As Integer, ByVal useCache As Boolean, ByVal columns As GridColumnsInfo) As Integer
			If View.OptionsView.NewItemRowPosition = NewItemRowPosition.None AndAlso rowHandle = CurrencyDataController.NewItemRow Then
				Return 0
			Else
				Return MyBase.CalcRowHeight(graphics, rowHandle, rowVisibleIndex, min, level, useCache, columns)
			End If
		End Function


		Public secondViewVScroll As ScrollBarPresence = ScrollBarPresence.Hidden

		Public Overrides ReadOnly Property VScrollBarPresence() As ScrollBarPresence
			Get
				If View.OptionsView.NewItemRowPosition = NewItemRowPosition.Bottom Then
				   Return secondViewVScroll
				Else
					Return MyBase.VScrollBarPresence
				End If
			End Get
		End Property

	End Class
End Namespace
